### PR TITLE
Add Renesas R-Car Setup documents

### PR DESCRIPTION
**Pull request by sparkgene**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+## Prerequisites
+
+- **Renesas Electronics Corporation R-Car S4 Reference Board/Spider**
+  - Spider board has eMMC and micro SD-card slot but the can't use simalutaneously.
+  - For using sdcard, please execute following two steps. The detail files can find in [R-Car S4 SDK Start Up Guide PKG](https://www.renesas.com/us/en/products/automotive-products/automotive-system-chips-socs/r-car-s4-automotive-system-chip-soc-car-servercommunication-gateway#design_development).
+   1. Please flash IPL which support booting from sdcard.
+      - It requires to build IPL with CA_LOAD_TYPE=1 option.
+      - For more details about building IPL, please refer to section 6.2.3 "How to build" in "RENESAS_ICUMX_IPL_for_R-Car_Gen4_Users_Manual_E.pdf"
+   1. Please change SW3 and SW6 on the CPU board.
+      - For more details, please refer to section 3.7.8.1 "Enable SD Card" in "R-Car S4_StartupGuide_x_x_x.pdf"
```
Comment (csxwin) [inline docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md:14]: Could you please provide the link to the Startup Guide?

File: docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+- **DHCP server with internet connection**
+  - In this guide, spider board requires DHCP server to connect internet.
```
Comment (csxwin) [inline docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md:16]: Does user need to configure anything for DHCP server?

File: docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+1. Insert the micro SD-card into the R-Car S4 Spider board’s micro SD-card slot.
+1. Connect an Ethernet cable.
+1. Connect develop machine to R-Car S4 Spider board USB port.
+   - USB port is depending on board revision(until B0-1st or since B0-2nd).
+   - For more detail, please refer to the section 2.1 "Linux BSP" in "R-Car S4_StartupGuide_x_x_x.pdf".
+
+   ![](./images/rcar-s4-spider.png)
+
+1. Use screen command on your develop machine terminal to veiw serial output.
```
[the remark this answers is not in the document]
Comment (sparkgene) [reply docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md:61]: Baud rate is `1843200`.
```
screen /dev/xxxxx 1843200
```

File: docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md (hunk 1 above)
Comment (sparkgene) [reply docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md:14]: The link is in line 9. The direct link to the document may change, so the link point to the overview page.

File: docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md (hunk 2 above)
Comment (sparkgene) [reply docs/dev-guide/edge-agent-dev-guide-renesas-rcar-s4.md:16]: No. If this confuse customer, we can change it as like NXP's document.
```
**Internet Router with Ethernet** — The S32G-VNP-RDB2 must be connected to an internet router via an Ethernet cable for internet connectivity. It is beyond the scope of this document to describe how this is achieved, but one possibility is to use a WiFi to Ethernet bridge and a smartphone acting as an internet hotspot.
```

**Comment by hefroy** [comment]
Please convert `docs/dev-guide/images/rcar-s4-spider.png` to JPG format - it is 1.4 MB at the moment, and should be around 170 KB when converted.

File: tools/renesas-rcar-s4/make-rootfs.sh [hunk 4]
```diff
@@ -0,0 +1,208 @@
+#!/bin/bash
+SCRIPT_DIR=$(cd `dirname $0` && pwd)
```
Comment (hefroy) [inline tools/renesas-rcar-s4/make-rootfs.sh:2]: Recommend to add `set -euo pipefail` to halt the script on the first error

Comment (sparkgene) [reply tools/renesas-rcar-s4/make-rootfs.sh:5]: added

**Comment by sparkgene** [comment]
Update the commented files

**Review by hefroy — APPROVED** [review]

**Pull request merged**